lib/int64.js

```diff
@@ -0,0 +1,335 @@
+        pos++;
+      }
+      const mask = (1 << (pos + 1)) - 1;
+      result[0] = (~(result[0] | mask)) | (result[0] & mask);
+      result[1] = ~result[1];
+    }
+    return result;
+  }
+
+  add(b) {
+    const tmp = this.value[0] + b.value[0];
+    this.value[0] = tmp;
+    this.value[1] += b.value[1] + (tmp > UINT32_MAX ? 1 : 0);
+    return this;
+  }
+
+  static add(a, b) {
+    return new Int64(a).add(b);
+  }
+
+  sub(b) {
+    let value = new Uint32Array(b.value);
+    if (value[0] !== 0 || value[1] !== 0) value = Int64._conversion(value);
+    const tmp = this.value[0] + value[0];
+    this.value[0] = tmp;
+    this.value[1] += value[1] + (tmp > UINT32_MAX ? 1 : 0);
+    return this;
+  }
+
+  static sub(a, b) {
+    return new Int64(a).sub(b);
+  }
+
+  static cmp(a, b) {
+    const aSign = a.value[1] >>> 31;
+    const bSign = b.value[1] >>> 31;
+    if (aSign !== bSign) return aSign ? -1 : 1;
+
+    if (a.value[1] > b.value[1]) return 1;
+    if (a.value[1] < b.value[1]) return -1;
+    if (a.value[0] === b.value[0]) return 0;
+    return a.value[0] > b.value[0] ? 1 : -1;
+  }
+
+  static _division(n, d) {
+    const zero = new Int64();
+    const one = new Int64(1);
+    if (Int64.cmp(d, zero) === 0) {
+      throw new RangeError('Int64: division by zero');
+    }
+    if (Int64.cmp(n, zero) === 0) return [zero, 0];
+    const cmp = Int64.cmp(d, n);
+    if (cmp === 0) return [one, zero];
+
+    const nSign = n.value[1] >>> 31;
+    let value = new Uint32Array(n.value);
+    if (nSign) value = Int64._conversion(value);
+    const dSign = d.value[1] >>> 31;
+    const dPositive = new Int64(d);
+    if (dSign) dPositive.value = Int64._conversion(dPositive.value);
+
+    const q = new Int64();
+    const r = new Int64();
+    for (let i = 63; i >= 0; i--) {
+      r.shiftLeft(1);
+      const valIndex = i >>> 5;
+      const nval = value[valIndex];
+      const ii = i < 32 ? i : i - 32;
+      r.value[0] = (r.value[0] & ~1) | ((nval & (1 << ii)) >>> ii);
+      if (Int64.cmp(r, dPositive) >= 0) {
+        r.sub(dPositive);
+        q.value[valIndex] |= 1 << ii;
+      }
+    }
+    q.value[1] &= INT32_MAX;
+    r.value[1] &= INT32_MAX;
+    if (nSign ^ dSign) q.value = Int64._conversion(q.value);
+    if (nSign) r.value = Int64._conversion(r.value);
+    return [q, r];
+  }
+
+  static div(a, b) {
+    return Int64._division(a, b)[0];
+  }
+
+  static mod(a, b) {
+    return Int64._division(a, b)[1];
+  }
+
+  static mult(a, b) {
+    const zero = new Int64();
+    const one = new Int64(1);
+    if (Int64.cmp(a, zero) === 0 || Int64.cmp(b, zero) === 0) return zero;
+    if (Int64.cmp(a, one) === 0) return new Int64(b);
+    if (Int64.cmp(b, one) === 0) return new Int64(a);
+
+    const aSign = a.value[1] >>> 63;
+    const bSign = b.value[1] >>> 63;
+    const result = new Int64();
+    const aPositive = new Int64(a);
+    let value = new Uint32Array(b.value);
+    if (aSign) aPositive.value = Int64._conversion(aPositive.value);
```

Comment:
https://github.com/metarhia/common/pull/203/files?utf8=%E2%9C%93&diff=unified#r227758289 Also applies here, I think there is better way of multiplying these than transforming them twice.